apps/passport-server/src/services/telegramService.ts

```diff
@@ -216,63 +308,93 @@ export class TelegramService {
+          );
+          return;
+        }
 
-      const admins = await ctx.getChatAdministrators();
-      const isAdmin = admins.some(
-        (admin) => admin.user.id === this.bot.botInfo.id
-      );
-      if (!isAdmin) {
-        await ctx.reply(
-          "Please add me as an admin to the telegram channel associated with your event."
+        const admins = await ctx.getChatAdministrators();
+        const isAdmin = admins.some(
+          (admin) => admin.user.id === this.bot.botInfo.id
         );
-        return;
-      }
+        if (!isAdmin) {
+          await ctx.reply(
+            "Please add me as an admin to the telegram channel associated with your event."
+          );
+          return;
+        }
 
-      const channelId = ctx.chat.id;
-      // TODO: Add menu of possible events
-      const eventName = ctx.match;
-      if (!eventName) {
+        const channelId = ctx.chat.id;
+        this.deleteEvents = false;
         await ctx.reply(
-          `Correct usage of this command is: /link <event_name>. Try: /link ProgCrypto (Internal Test)`
+          `Checking linked status of this chat... (id: ${channelId})`
         );
-        return;
-      }
 
-      await ctx.reply("Your telegram channel id is " + ctx.chat.id);
+        await this.loadEvents(this.context.dbPool);
+        await ctx.reply("Link options", {
+          reply_markup: eventsMenu
+        });
+      } catch (error) {
+        await ctx.reply(`Error linking. Check server logs for details`);
+        logger(`[TELEGRAM] ERROR`, error);
+      }
+    });
 
-      // TODO: Remove fetchPretixEventInfoByName
-      // https://github.com/proofcarryingdata/zupass/issues/662
+    this.bot.command("unlink", async (ctx) => {
       try {
-        const eventInfo = await fetchPretixEventInfoByName(
-          this.context.dbPool,
-          eventName
-        );
-        if (!eventInfo) throw new Error(`Failed to fetch event ${eventName}`);
+        await ctx.reply(`Checking you have permission to unlink...`);
+        if (ctx.chat?.type === "private") {
+          await ctx.reply(
+            "To get you started, can you please add me as an admin to the telegram channel associated with your event? Once you are done, please ping me again with /setup in the channel."
+          );
+          return;
+        }
 
-        await insertTelegramEvent(
-          this.context.dbPool,
-          eventInfo.pretix_events_config_id,
-          channelId
-        );
-        logger(
-          `[TELEGRAM] linked event ${eventName} to group ${ctx.chat.title}`
-        );
-        await ctx.reply(
-          `Linked ${ctx.chat.title} (id: ${channelId}) with event ${eventName}`
+        const admins = await ctx.getChatAdministrators();
+        const isAdmin = admins.some(
+          (admin) => admin.user.id === this.bot.botInfo.id
         );
+        if (!isAdmin) {
```

Comment:
@rrrliu this duplicate logic (also appears in `/link`) will be improved in next admin bot PR.